crates/shared/src/price_estimation/native/oneinch.rs

```diff
@@ -105,16 +125,36 @@ async fn update_prices(
         .text()
         .await
         .context("Failed to fetch Native 1Inch prices")?;
-    let result = serde_json::from_str::<Response>(&response)
-        .with_context(|| format!("Failed to parse Native 1inch prices from {response:?}"))?;
-    Ok(result.0)
+    let prices = serde_json::from_str::<Response>(&response)
+        .with_context(|| format!("Failed to parse Native 1inch prices from {response:?}"))?
+        .0;
+
+    let token_infos = token_info
+        .get_token_infos(&prices.keys().copied().collect::<Vec<_>>())
```

Comment:
why do you need to copied it if you are passing it by ref?

Reply:
that's the type required by `get_token_infos` function.

Reply:
Yeah, I have to pass a reference to a slice but it has to be `&[H160]` not `&[&H160]`.